### ISSUE TITLE
Enhancement:  to_json and read_json for DataFrame should have option to output/parse values by column

### ISSUE THREAD

**Issue by Dr-Irv** [pandas-dev/pandas#17220]
#### Code Sample, a copy-pastable example if possible

```python
ll = [(1, "a", 4), (2, "b", 5), (3, "c", 6)]
df = pd.DataFrame(ll, columns=["one", "two", "three"])
print(df.to_json(orient='split'))
print(df['one'].to_json(orient='split'))
```
#### Problem description

The output for this code is:
```
{"columns":["one","two","three"],"index":[0,1,2],"data":[[1,"a",4],[2,"b",5],[3,"c",6]]}
{"name":"one","index":[0,1,2],"data":[1,2,3]}
```
Note that the values for the `DataFrame` are _by row_, whereas the values for the `Series` are _by column_.

The issue with the `DataFrame` version is that parsing the values in a type-safe language requires you to handle each element of the array as a different type, while using the Series format, the values are all of one type.

So I'd like to propose that we add an `orient='columns'` for `DataFrame`.  In this case, it would look just like the `'split'` option, but instead of a `'data'` field, it would have a `'cdata'` field, where you'd have a list of values for each column of the DataFrame. 

Note that this can almost be done by using `df.T.to_json(orient='split')`, but that reverses the meaning of `'index'` and `'columns'` in the resulting JSON.

Finally, I presume we'd want to change the implementation of `read_json()` and `to_json()`.

#### Expected Output
```python
print(df.to_json(orient='columns''))
```
would produce
```
{"columns":["one","two","three"],"index":[0,1,2],"cdata":[[1,2,3],["a","b","c"],[4,5,6]]}
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.1.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.20.3
pytest: None
pip: 9.0.1
setuptools: 27.2.0
Cython: None
numpy: 1.12.1
scipy: 0.19.0
xarray: None
IPython: 6.0.0
sphinx: None
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.0.0
openpyxl: None
xlrd: 1.0.0
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: None

</details>

**Comment by jreback** [comment]
this is not an issue for ``.to_json()`` you can simply specify the columns you want a-priorit.

For reading you can do something like this: http://pandas.pydata.org/pandas-docs/stable/io.html#normalization

note this is very unlikely to happen in pandas before 2.0 as the reader is not friendly to this kind of logic. however if some enteprising soul wants to implement would take a PR.

**Comment by Dr-Irv** [comment]
@jreback Not sure what you mean by "specify the columns you want a-priori".  I know that I can do something like `[df[s].to_json(orient = 'split') for s in df.columns ]` to have each column as a single vector of values in the JSON, but you seem to be referring to something else.

If I get some time, I might look into doing a PR.

**Comment by ashukla411** [comment]
I agree if you want only selective keys from the json without actually loading into a dataframe format. It will save some memory issues that I have too when I read_json a very large dataset.
Right now I have writtien a function to loop over json keys and collect only necessary ones.

**Comment by jbrockmendel** [comment]
Does orient="columns" not handle this?

**Comment by Dr-Irv** [comment]
Almost.  Issue there is that for each column, the index is repeated.  Using the above example:
```python
df.to_json(orient='columns')
```
produces
```text
'{"one":{"0":1,"1":2,"2":3},"two":{"0":"a","1":"b","2":"c"},"three":{"0":4,"1":5,"2":6}}'
```
Note the repetition of the index in each column.  With a DF with many columns, that's a lot of repetition.   Also, the names of the index and columns are lost.

I'd like something like the `dict` produced by `df.to_dict(orient="tight")`, but where the values are column oriented.
```python
df.to_dict(orient="tight")
```
which produces
```text
{'index': [0, 1, 2], 'columns': ['one', 'two', 'three'], 'data': [[1, 'a', 4], [2, 'b', 5], [3, 'c', 6]], 'index_names': [None], 'column_names': [None]}
```

Note for the `values` in `to_dict()` and `to_json()`, I don't know why we output that in row order and not column order.
